indico/modules/events/registration/apple_pass.py

```diff
@@ -0,0 +1,133 @@
+
+        Rewritten to use cryptography library instead of M2Crypto
+        The manifest is the file containing a list of files included in the pass file (and their hashes).
+        """
+        # if the certificate has already been converted to Byte, just use it as is.
+        cert = x509.load_pem_x509_certificate(certificate.encode('UTF-8')) if isinstance(certificate, str) else (
+            certificate)
+        private_key = serialization.load_pem_private_key(key.encode('UTF-8'), password=password.encode('UTF-8'))
+        wwdr_cert = x509.load_pem_x509_certificate(Path(wwdr_certificate).read_bytes())
+        options = [pkcs7.PKCS7Options.DetachedSignature]
+        return (
+            pkcs7.PKCS7SignatureBuilder()
+            .set_data(manifest)
+            .add_signer(cert, private_key, hashes.SHA256())
+            .add_certificate(wwdr_cert)
+            .sign(serialization.Encoding.DER, options)
+        )
+
+
+class ApplePassManager:
+    """Apple Pass ticketing manager.
+
+    This class handles the integration with the Apple Pass ticketing API.
+    """
+
+    def __init__(self, event):
+        self.event = event
+        self.settings = self.event.category.effective_apple_pass_config if not self.event.is_unlisted else None
+        self.cert = None
+        self.static_path = os.path.join(current_app.root_path, 'web', 'static')
+
+    @property
+    def configured(self):
+        return config.ENABLE_APPLE_PASS and self.settings is not None
+
+    def build_ticket_object(self, registration):
+        from indico.modules.categories.controllers.util import make_format_event_date_func
+        ticket = EventTicket()
+        ticket.addPrimaryField('event-title', self.event.title, 'Event')
+        event_date = make_format_event_date_func(self.event.category)(self.event)
+        ticket.addSecondaryField('event-date', event_date, 'Date')
+        if self.event.address and self.event.has_location_info:
+            venue_name = (f'{self.event.room_name} ({self.event.venue_name})'
+                          if self.event.room_name and self.event.venue_name
+                          else (self.event.venue_name or self.event.room_name))
+            ticket.addSecondaryField('event-venue', venue_name, 'Venue')
+        ticket.addAuxiliaryField('registration-name', registration.full_name, 'Name')
+        ticket.addAuxiliaryField('registration-email', registration.email, 'Email')
+        ticket.addBackField('back-registration-name', registration.full_name, 'Name')
+        ticket.addBackField('back-event-venue', venue_name, 'Venue')
+        ticket.addBackField('back-ticket-number', f'#{registration.friendly_id}', 'Ticket number')
+        ticket.addBackField('back-event-date', event_date, 'Date')
+        ticket.addBackField('back-registration-email', registration.email, 'Email')
+        ticket.addBackField('back-event-url', f'<a href="{self.event.external_url}">Indico event page</a>', 'Link')
```

Comment:
How about just "Event page"?

```suggestion
        ticket.addBackField('back-event-url', f'<a href="{self.event.external_url}">Event page</a>', 'Link')
```

Reply:
Done.
(Well, the Event may have a separate web page.)